test/test_ops.py

```diff
@@ -801,30 +824,35 @@ def out_fn(output):
                             return sample.output_process_fn_grad(output)
                         return output
 
+                    def get_sample():
+                        return clone_input_helper(sample.input) if op.name[-1] == '_' else sample.input
```

Comment:
I don't think this helper is needed if the variant can never be inplace

Reply:
hmm do we know that ? i guess i'll add an assertion